sdk/kiosk/src/client/kiosk-client.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiClient } from '@mysten/sui.js/src/client';
+import { fetchKiosk, getOwnedKiosks } from '../query/kiosk';
+import {
+	type FetchKioskOptions,
+	KioskData,
+	OwnedKiosks,
+	Network,
+	type KioskClientOptions,
+} from '../types';
+import { TransferPolicyRule, PERSONAL_KIOSK_RULE_ADDRESS, rules } from '../constants';
+import { queryTransferPolicy } from '../query/transfer-policy';
+
+/**
+ * A Client that allows you to interact with kiosk.
+ * Offers utilities to query kiosk, craft transactions to edit your own kiosk,
+ * purchase, manage transfer policies, create new kiosks etc.
+ */
+export class KioskClient {
+	client: SuiClient;
+	network: Network;
+	rules: TransferPolicyRule[];
+
+	constructor(options: KioskClientOptions) {
+		this.client = options.client;
+		this.network = options.network;
+		this.rules = rules; // add all the default rules.
+	}
+
+	/// Querying
+
+	/**
+	 * Get an addresses's owned kiosks.
+	 * @param address The address for which we want to retrieve the kiosks.
+	 * @returns An Object containing all the `kioskOwnerCap` objects as well as the kioskIds.
+	 */
+	async getOwnedKiosks({ address }: { address: string }): Promise<OwnedKiosks> {
+		const personalPackageId = PERSONAL_KIOSK_RULE_ADDRESS[this.network];
+
+		return getOwnedKiosks(this.client, address, {
+			personalKioskType: personalPackageId
+				? `${personalPackageId}::personal_kiosk::PersonalKioskCap`
+				: '',
+		});
+	}
+
+	/**
+	 * Fetches the kiosk contents.
+	 * @param kioskId The ID of the kiosk to fetch.
+	 * @param options Optioal
+	 * @returns
+	 */
+	async getKiosk({ id, options }: { id: string; options: FetchKioskOptions }): Promise<KioskData> {
```

Comment:
this could probably be `getKiosk({ id, ...options}: { id: string } & FetchKioskOptions)` or `getKiosk(id, options: FetchKioskOptions = {})`

It looks like options isn't currently optional

Reply:
I actually ended up going with `options?`, to preserve the same structure as our TS SDK (id, options keys)